paasta_tools/cli/cmds/spark_run.py

```diff
@@ -538,6 +577,12 @@ def get_spark_app_name(original_docker_cmd: Union[Any, str, List[str]]) -> str:
         spark_app_name = "paasta_spark_run"
 
     spark_app_name += f"_{get_username()}"
+    if enable_k8s_autogen:
+        document = POD_TEMPLATE.format(spark_app_name=spark_app_name)
+        parsed_pod_template = yaml.load(document)
+        with open(POD_TEMPLATE_PATH, "w") as f:
+            yaml.dump(parsed_pod_template, f)
```

Comment:
Just a nit: do we want to keep the `yaml.load/yaml.dump` parsing, instead of writing to file directly, just to verify that the format is YAML-compatible?

Reply:
yeah I think keep the laod/dump is good to make sure that the file format is good